### PR TITLE
LPD-27473 Internal server error when sending the Object Entry `status` field without the `code` subfield

### PR DESCRIPTION
**Pull request by dannielraposo**
**What's changed?:**
- The _code_ subfield is now required within the _status_ field
- Tests have been added to validate server's response

**How to test it?:**
- Execute both of the following curl commands:
```
curl -X POST -u test@liferay.com:test -H 'Content-type: application/json' -d '{
            "enableCategorization": true,
            "externalReferenceCode": "STUDENT",
            "label": {
                "en_US": "Student"
            },
            "modifiable": true,
            "name": "Student",
            "objectFields": [
                {
                    "DBType": "String",
                    "businessType": "Text",
                    "externalReferenceCode": "STUDENT_NAME",
                    "indexed": true,
                    "indexedAsKeyword": false,
                    "indexedLanguageId": "en_US",
                    "label": {
                        "en_US": "studentName"
                    },
                    "name": "studentName",
                    "required": false,
                    "state": false,
                    "system": false,
                    "type": "String"
                }
            ],
            "panelCategoryKey": "",
            "parameterRequired": false,
            "pluralLabel": {
                "en_US": "Students"
            },
            "portlet": false,
            "scope": "company",
            "status": {
                "code": 0
            },
            "system": false
        }' http://localhost:8080/o/object-admin/v1.0/object-definitions
```

```
curl -X POST \
     -u 'test@liferay.com:test' \
     http://localhost:8080/o/c/students \
     -H 'Content-type: application/json' \
     -d '{
            "studentName": "student 1",
            "status" : {
               "label" : "approved"
            }
         }'
```

A 400 HTTP error would be returned because the code subfield is necessary.

See the following Jira Ticket: [LPD-27473](https://liferay.atlassian.net/browse/LPD-27473)

### CODE REVIEW

**Comment by dannielraposo** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-api/derived-config.txt
Comment (4lejandrito) [inline modules/apps/object/object-rest-api/derived-config.txt:1]: Delete this file.

File: lib/portal/wstx.jar
Comment (4lejandrito) [inline lib/portal/wstx.jar:1]: Delete this file.

File: modules/apps/object/object-rest-api/src/main/java/com/liferay/object/rest/dto/v1_0/Status.java
Comment (4lejandrito) [inline modules/apps/object/object-rest-api/src/main/java/com/liferay/object/rest/dto/v1_0/Status.java:1]: The generated code must be inside a separate commit.

File: modules/apps/object/object-rest-impl/derived-config.txt
Comment (4lejandrito) [inline modules/apps/object/object-rest-impl/derived-config.txt:1]: Delete this file.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java
Comment (4lejandrito) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:1]: Also include the tests in a separate commit.

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java [hunk 1]
```diff
@@ -7158,6 +7158,43 @@ public void testPostCustomObjectEntryWithAttachmentObjectField()
 			_siteScopedObjectDefinition1);
 	}
 
+	@Test
+	public void testPostCustomObjectEntryWithCodeSubfield()
+		throws Exception {
+
+
+		//Without status field
```
Comment (4lejandrito) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java:7166]: Add an space after the `//` and a new empty line below the comment.

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Good job on your first PR!
